src/settings.rs

```diff
@@ -85,16 +97,18 @@ pub async fn restore(req: Request<Body>) -> Result<Response<Body>, String> {
 
 	let form = url::form_urlencoded::parse(query).collect::<HashMap<_, _>>();
 
-	let names = vec!["theme", "front_page", "layout", "wide", "comment_sort", "post_sort", "show_nsfw", "subscriptions"];
-
-	let path = match form.get("redirect") {
-		Some(value) => format!("/{}/", value),
+	let mut path = match form.get("redirect") {
+		Some(value) => format!("/{}", value.replace("%26", "&").replace("%23", "#")),
 		None => "/".to_string(),
 	};
 
+	if !path.contains('#') {
```

Comment:
This is to keep old behavior of ending links with `/` but only in case that link does not contain `#id` part.

Links with scrolling to element `id` do not work if there is `/` appended at the end and I was not sure if all use cases will work correctly without trailing `/`. This way processing of old links will have exact same result like before this change. 

This `if` won't be needed if trailing `/` is not necessary or in case global notification banner will be used then format on line 101 can contain trailing `/` and also this `if` could be removed.

Reply:
That makes perfect sense, thanks.

Previously, Libreddit required trailing slashes in order to properly route users. Though, since the [Hyper rewrite](https://github.com/spikecodes/libreddit/commit/b14b4ff551f24eec08bf7647c88dcea85b6a0d73), this is no longer necessary; the links https://libredd.it/wiki/markdown and https://libredd.it/wiki/markdown/ will be matched under the same route now.

Therefore, I don't believe this `if` statement is needed as a trailing `/` is no-longer necessary.